File: Sample/CDCodabarGenerator/ViewController.m
//
//  ViewController.m
//  CDCodabarGenerator
//
//  Created by Cole Dunsby on 2014-04-17.
//  Copyright (c) 2014 Cole Dunsby. All rights reserved.
//

#import "ViewController.h"
#import "CDCodabarView.h"

@interface ViewController ()

@property (nonatomic, strong) CDCodabarView *codabarView;

- (UIColor *)randomColor;
- (NSString *)randomBarcode;

@end

@implementation ViewController

#pragma mark - View Lifecycle

- (void)viewDidLoad
{
    [super viewDidLoad];
    
    self.codabarView = [[CDCodabarView alloc] initWithCode:@"123456789"
                                                 startChar:'A'
                                                  stopChar:'B'];
    
    [self changeColorsButtonPressed:nil];
    self.codabarView.hideCode = NO;
    
    self.codabarView.center = self.view.center;
    
    [self.view addSubview:self.codabarView];
}

#pragma mark - Helper Methods

- (UIColor *)randomColor
{
    CGFloat hue = (arc4random() % 256 / 256.0);
    CGFloat saturation = (arc4random() % 128 / 256.0) + 0.5;
    CGFloat brightness = (arc4random() % 128 / 256.0) + 0.5;
    
    return [UIColor colorWithHue:hue saturation:saturation brightness:brightness alpha:1.0];
}

- (NSString *)randomBarcode
{
    NSMutableString *barcode = [[NSMutableString alloc] init];
    
    for (int i = 0; i < 9; i++)
    {
        [barcode appendString:[NSString stringWithFormat:@"%i", arc4random() % 10]];
    }
    
    return barcode;
}

#pragma mark - IBActions

- (IBAction)heightSliderValueChanged:(id)sender
{
    self.codabarView.barHeight = self.heightSlider.value;
    
    self.heightValueLabel.text = [NSString stringWithFormat:@"%.1f", self.heightSlider.value];
}

- (IBAction)widthSliderValueChanged:(id)sender
{
    self.codabarView.barWidth = self.widthSlider.value;
    
    self.widthValueLabel.text = [NSString stringWithFormat:@"%.1f", self.widthSlider.value];
}

- (IBAction)paddingSliderValueChanged:(id)sender
{
    self.codabarView.padding = self.paddingSlider.value;
    
    self.paddingValueLabel.text = [NSString stringWithFormat:@"%.1f", self.paddingSlider.value];
}

- (IBAction)borderSwitchValueChanged:(id)sender
{
    self.codabarView.layer.borderWidth = (self.codabarView.layer.borderWidth == 1.0) ? 0.0 : 1.0;
    
    for (UIView *view in self.codabarView.subviews)
    {
        view.layer.borderWidth = (view.layer.borderWidth == 1.0) ? 0.0 : 1.0;
    }
}

- (IBAction)changeColorsButtonPressed:(id)sender
{
    self.codabarView.barColor = [self randomColor];
    self.codabarView.textColor = [self randomColor];
}

- (IBAction)changeBarcodeButtonPressed:(id)sender
{
    [self.codabarView setCode:[self randomBarcode]
                    startChar:self.codabarView.startChar
                     stopChar:self.codabarView.stopChar];
    
    self.codabarView.center = self.view.center;
}

@end

File: Example/CDCodabarView_Example/ViewController.swift
//
//  ViewController.swift
//  CDCodabarViewSample
//
//  Created by Cole Dunsby on 2015-12-21.
//  Copyright © 2015 Cole Dunsby. All rights reserved.
//

import UIKit
import CDCodabarView

class ViewController: UIViewController {

    @IBOutlet weak var heightLabel: UILabel!
    @IBOutlet weak var widthLabel: UILabel!
    @IBOutlet weak var paddingLabel: UILabel!
    @IBOutlet weak var codabarView: CDCodabarView!
    @IBOutlet weak var heightConstraint: NSLayoutConstraint!
    @IBOutlet weak var widthConstraint: NSLayoutConstraint!
    
    // MARK: - View Lifecycle
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        /* Adding barcode programmatically
        let barcode = CDCodabarView(frame: CGRect(x: 0, y: 0, width: 200, height: 100))
        barcode.code = "A1234B"
        barcode.backgroundColor = .whiteColor()
        view.addSubview(barcode)
        */
    }
    
    // MARK: - Private Functions
    
    func randomColor() -> UIColor {
        return UIColor(red: CGFloat(drand48()), green: CGFloat(drand48()), blue: CGFloat(drand48()), alpha: 1.0)
    }
    
    func randomBarcode() -> String {
        let scalars = "A".unicodeScalars
        let unicode = Int(scalars[scalars.startIndex].value)
        
        let startChar = Character(UnicodeScalar(unicode + Int(arc4random_uniform(4))))  // A, B, C or D
        let stopChar = Character(UnicodeScalar(unicode + Int(arc4random_uniform(4))))   // A, B, C or D
        
        let barcodeLength = Int(arc4random_uniform(6)) + 5  // 5 - 10
        var barcode = ""
        
        for var i = 0; i < barcodeLength; i++ {
            barcode += String(arc4random_uniform(10))   // 0 - 9
        }
        
        return String(startChar) + barcode + String(stopChar)
    }
    
    // MARK: - IBActions
    
    @IBAction func heightSliderValueChanged(sender: UISlider) {
        heightConstraint.constant = CGFloat(sender.value)
        heightLabel.text = "\(Int(sender.value))"
        
        codabarView.setNeedsDisplay()
    }
    
    @IBAction func widthSliderValueChanged(sender: UISlider) {
        widthConstraint.constant = CGFloat(sender.value)
        widthLabel.text = "\(Int(sender.value))"
        
        codabarView.setNeedsDisplay()
    }
    
    @IBAction func paddingSliderValueChanged(sender: UISlider) {
        codabarView.padding = CGFloat(sender.value)
        paddingLabel.text = String(format: "%.1f", sender.value)
    }
    
    @IBAction func hideCodeSwitchValueChanged(sender: UISwitch) {
        codabarView.hideCode = sender.on
    }
    
    @IBAction func changeColorsButtonTapped(sender: UIButton) {
        codabarView.barColor = randomColor()
        codabarView.textColor = randomColor()
    }
    
    @IBAction func changeBarcodeButtonTapped(sender: UIButton) {
        codabarView.code = randomBarcode()
    }

}

